src/crepe/facade/SDLContext.cpp

```diff
@@ -383,101 +276,124 @@ ivec2 SDLContext::get_size(const Texture & ctx) {
 std::vector<SDLContext::EventData> SDLContext::get_events() {
 	std::vector<SDLContext::EventData> event_list;
 	SDL_Event event;
-
-	// Handle general SDL events
+	const CameraAuxiliaryData & cam = this->cam_aux_data;
 	while (SDL_PollEvent(&event)) {
 		ivec2 mouse_pos;
 		mouse_pos.x = (event.button.x - cam.bar_size.x) / cam.render_scale.x;
 		mouse_pos.y = (event.button.y - cam.bar_size.y) / cam.render_scale.y;
 		switch (event.type) {
 			case SDL_QUIT:
-				event_list.push_back({SDLContext::EventType::SHUTDOWN, {}, {}, {}});
+				event_list.push_back({.event_type = SDLContext::EventType::SHUTDOWN});
 				break;
-			case SDL_KEYDOWN:
-				event_list.push_back(
-					{SDLContext::EventType::KEYDOWN,
-					 {sdl_to_keycode(event.key.keysym.scancode), event.key.repeat != 0},
-					 {},
-					 {}});
+			case SDL_KEYDOWN: {
+				this->update_keyboard_state();
+				EventData transfer_event;
+				transfer_event.event_type = SDLContext::EventType::KEYDOWN;
+				transfer_event.data.key_data = KeyData{
+					.key = sdl_to_keycode(event.key.keysym.scancode),
+					.key_repeat = event.key.repeat != 0,
+				};
+				event_list.push_back(transfer_event);
```

Comment:
Dit kan gewoon geschreven worden als
```cpp
event_list.push_back(EventData{
	.event_type = SDLContext::EventType::KEYDOWN,
	.data = {
		.key_data = {
			.key = sdl_to_keycode(event.key.keysym.scancode),
			.key_repeat = event.key.repeat != 0,
		},
	},
});
```
als je de `EventData` struct definitie aanpast in `SDLContext.h` naar
```cpp
struct EventData {
	SDLContext::EventType event_type = SDLContext::EventType::NONE;
	union {
		KeyData key_data;
		MouseData mouse_data;
		WindowData window_data;
	} data;
};
```